maker/src/connection.rs

```diff
@@ -453,76 +453,81 @@ impl Actor {
             .send_async_safe(cfd::TakerConnected { id: identity })
             .await;
 
-        let mut tasks = Tasks::default();
-        tasks.add_fallible(
-            {
-                let this = this.clone();
-
-                async move {
-                    loop {
-                        let msg = read
-                            .try_next()
-                            .await
-                            .context("Failed to read from socket")?
-                            .context("End of stream")?;
-
-                        this.send(cfd::FromTaker {
-                            peer_id,
-                            taker_id: identity,
-                            msg,
-                        })
-                        .await
-                        .context(
-                            "we are not connected to ourselves, this should really not happen",
-                        )?;
-                    }
-                }
-            },
-            {
-                let this = this.clone();
-
-                move |error| async move {
-                    let _ = this
-                        .send(ReadFail {
-                            taker_id: identity,
-                            error,
-                        })
-                        .await;
-                }
-            },
-        );
+        tracing::debug_span!("Test span container").in_scope(|| {
```

Comment:
@Restioson Could the reason why we see these long-lasting spans be that this task is an endless loop? I would expect it to never resolve (unless we encounter an error).

Reply:
Yes, I think so. However, the main issue at hand is that these tasks shouldn't extend the duration of the calling span at all. For some reason, they get attached as children and this is the issue.